server/api/discrepancies.py

```diff
@@ -195,32 +229,61 @@ def get_ballot_comparison_audit_discrepancies_by_jurisdiction(
         )
     )
 
+    show_discrepancies_by_jurisdiction = (
+        show_ballot_comparison_discrepancies_by_jurisdiction(election)
+    )
+
     for contest in election.contests:
         audited_results = sampled_ballot_interpretations_to_cvrs(contest)
         reported_results = cvrs_for_contest(contest)
         for ballot_id, audited_result in audited_results.items():
-            audited_cvr = audited_result["cvr"]
-            reported_cvr = reported_results.get(ballot_id)
-            vote_deltas = ballot_vote_deltas(contest, reported_cvr, audited_cvr)
-            if not vote_deltas or isinstance(vote_deltas, str):
-                continue
-
-            # CVRs are guaranteed to be non-null due to ballot_vote_deltas() checks
-            assert isinstance(audited_cvr, dict)
-            audited_votes = audited_cvr.get(contest.id, {})
-            assert isinstance(reported_cvr, dict)
-            reported_votes = reported_cvr.get(contest.id, {})
             if ballot_id in sampled_ballot_id_to_jurisdiction_id:
```

Comment:
The core diff for this block is really just lines 240 to 242 of the updated code. The `if ballot_id in sampled_ballot_id_to_jurisdiction_id:` was moved higher up as I needed `jurisdiction_id` to index `show_discrepancies_by_jurisdiction`.

That being said, it seems like it would be safe to drop that check, as I think `jurisdiction_id` not being present in `sampled_ballot_id_to_jurisdiction_id` would be a data integrity issue. Does that sound right to you?

Reply:
Hmm, it looks like `ballot_id` is coming from `sampled_ballot_interpretations_to_cvrs`, which by my read doesn't filter by round, so it could include ballots from earlier rounds that aren't in this round. `sampled_ballot_id_to_jurisdiction_id` only includes ballots sampled in this round. So that might create a case where `ballot_id` is not in `sampled_ballot_id_to_jurisdiction_id`.

But I could be misreading - could try removing it to see if a test fails (if it's a necessary guard, we should cover it with a test)

Reply:
Oh right. I forgot that `sampled_ballot_interpretations_to_cvrs` includes ballots from other rounds. Thank you

Reply:
Added another call to `/discrepancies` at the end of `test_ballot_comparison_two_rounds` to test that code path. Now when we hit the endpoint after round 2 is completed, we skip over the round 1 audit ballots, as expected.